Misc doc changes

kianmeng opened the pull request:
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

cpjolicoeur commented on README.md:
```diff
@@ -217,3 +220,9 @@ config :torch,
+
+Copyright (c) 2016 Infinite Red, Inc.
+
+This software is released under the [MIT License](./LICENSE.md).
```
This whole added section is not needed (somewhat incorrect) and can/should be removed.

kianmeng replied:
You're right about the incorrectness, it supposed to be:

```
Original work Copyright (c) 2016 Infinite Red, Inc. 
Modifed work Copyright (c) 2019 MojoTech, LLC. 
```

I've removed it as requested.

cpjolicoeur commented on mix.exs:
```diff
@@ -1,17 +1,17 @@
       name: "Torch",
-      description: "Rapid admin generator for Phoenix",
-      source_url: "https://github.com/mojotech/torch",
-      homepage_url: "https://github.com/mojotech/torch",
```
Why are we removing this line for `homepage_url`?

kianmeng replied:
My bad. Added it back.

cpjolicoeur commented:
@kianmeng Thanks for the pull request.  I have a few questions and change requests though.

First, I'm not sure what you mean by "this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library".  This is already the case, is it not?  We generated updated documentation
and upload it every time we release a new version of the library.  I'm not sure how these changes ensures that in a way beyond what we already have and do?

Second, you made some additional changes to other files that seem unrelated to the description.  For example, some changes to the `.formatter.exs` file, `.gitignore`, changing the `LICENSE` file to markdown.  Things like that are unrelated and we'd want to drop them from the commit and this pull request before merging.  

Also, some of the formatting changes made to the README file itself are unneeded.  I'm not very interested in adding badges for the code coverage amount, or the last commit information.  Some indentation and whitespace changes within the README content as well are not necessary as well.

kianmeng committed:
Misc doc changes

Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

kianmeng commented:
> @kianmeng Thanks for the pull request. I have a few questions and change requests though.

@cpjolicoeur Thanks for the feedback and letting me know your preference.

> First, I'm not sure what you mean by "this commit ensures the generated
> HTML doc for HexDocs.pm will become the source of truth for this Elixir
> library". This is already the case, is it not? We generated updated documentation
> and upload it every time we release a new version of the library. I'm not sure how these changes ensures that in a way beyond what we already have and do?

Basically all related documents in the repository like `CHANGELOG.md`, `CODE_OF_CONDUCT.md`, and `LICENSE` can be accessed through HexDocs.

> Second, you made some additional changes to other files that seem unrelated to the description. For example, some changes to the `.formatter.exs` file, `.gitignore`, changing the `LICENSE` file to markdown. Things like that are unrelated and we'd want to drop them from the commit and this pull request before merging.

Reverted these changes. `LICENSE` was originally renamed to `LICENSE.md` for consistency.

> Also, some of the formatting changes made to the README file itself are unneeded. I'm not very interested in adding badges for the code coverage amount, or the last commit information. Some indentation and whitespace changes within the README content as well are not necessary as well.

Likewise, changes reverted.

cpjolicoeur committed:
Move mix config keys to proper sections

* https://hex.pm/docs/publish

cpjolicoeur commented:
Thanks @kianmeng for this Pull Request.

I made some small changes to your README updates to make sure keys were set in the proper sections in the mix file.
